components/brave_wallet/browser/eth_data_parser.cc

```diff
@@ -67,6 +67,7 @@ bool GetTransactionInfoFromData(const std::string& data,
        {"0x095ea7b3", mojom::TransactionType::ERC20Approve},
        {"0x23b872dd", mojom::TransactionType::ERC721TransferFrom},
        {"0x42842e0e", mojom::TransactionType::ERC721SafeTransferFrom},
+       {"0xf242432a", mojom::TransactionType::ERC1155SafeTransferFrom},
```

Comment:
I think we should also support getting tx_args and tx_params parsed for UI?
See below for how we parsed for ERC20 and ERC721 transfers.

Reply:
Sounds good to me.  I added similar logic.  It does slightly different checks than what we do for ERC20, and ERC721.  The reason for the difference is that the last parameter is bytes, which can be of arbitrary length.  So can't really check if there is any 'leftover'.

Reply:
Sounds good to me. I added similar logic. It does slightly different checks than what we do for ERC20, and ERC721. The reason for the difference is that the last parameter is bytes, which can be of arbitrary length. So can't really check if there is any 'leftover'.